client/unsubscribe_transaction.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"fmt"
+	"strings"
+
+	msgs "github.com/energomonitor/bisquitt/messages"
+	"github.com/energomonitor/bisquitt/transactions"
+)
+
+type unsubscribeTransaction struct {
+	*transaction
+}
+
+func newUnsubscribeTransaction(client *Client, msgID uint16) *unsubscribeTransaction {
+	tLog := client.log.WithTag(fmt.Sprintf("UNSUBSCRIBE(%d)", msgID))
+	tLog.Debug("Created.")
+	return &unsubscribeTransaction{
+		transaction: &transaction{
+			RetryTransaction: transactions.NewRetryTransaction(
+				client.groupCtx, client.cfg.RetryDelay, client.cfg.RetryCount,
+				func(lastMsg interface{}) error {
+					tLog.Debug("Resend.")
+					return client.send(lastMsg.(msgs.Message))
+				},
+				func() {
+					tLog.Debug("Deleted.")
+					client.transactions.Delete(msgID)
+				},
+			),
+			client: client,
+			log:    tLog,
+		},
+	}
+}
+
+func (t *unsubscribeTransaction) Unsuback(_ *msgs.UnsubackMessage) {
+	var topicName string
+	unsubscribe := t.Data.(*msgs.UnsubscribeMessage)
+
+	switch unsubscribe.TopicIDType {
+	case msgs.TIT_STRING:
+		topicName = string(unsubscribe.TopicName)
+
+		t.log.Debug(`Topic "%s" unsubscribed`,
+			topicName,
+		)
+		t.client.registeredTopicsLock.Lock()
+		delete(t.client.registeredTopics, topicName)
+		t.client.registeredTopicsLock.Unlock()
```

Comment:
I don't understand this. Why to unregister topic on unsubscribe?

Reply:
For the record: we have reached an agreement that deleting a registered topic is not correct behavior for the fact that MQTT-SN doesn't mention this. I have deleted the lines in question.